pages/map.py

```diff
@@ -57,45 +57,78 @@ def create_lines_map(trips_group_by_user_id, user_id_list):
     return fig
 
 
-def create_heatmap_fig(data):
+def get_map_coordinates(trips_group_by_user_mode, user_mode_list):
+    coordinates = {'lat': [], 'lon': [], 'color':[]}
+    for user_mode in user_mode_list:
+        color = trips_group_by_user_mode[user_mode]['color']
+        trips = trips_group_by_user_mode[user_mode]['trips']
+
+        for trip in trips:
+            coordinates['lon'].append(trip['start_coordinates'][0])
+            coordinates['lon'].append(trip['end_coordinates'][0])
+            coordinates['lat'].append(trip['start_coordinates'][1])
+            coordinates['lat'].append(trip['end_coordinates'][1])
+            coordinates['color'].extend([color,color])
```

Comment:
Can you clarify why you picked this data structure, which requires you to iterate over all the trips? Also it seems weird that you concatenate the start and end longitudes and latitudes separately. Are you sure this is the expected input? Where is the related documentation?

Reply:
Note to self: this is copied over from
https://github.com/e-mission/op-admin-dashboard/pull/91/files#diff-eed798ce43b7a089fbd9491f30276d8779a357c64583d343e17b025d0c1559eeL271

I would still like to see some validation of this now that we are not in a hurry to get the admin dashboard merged.

Reply:
@achasmita I don't think this has been addressed yet. Why do you pick this data structure which seems like it has a lot of duplicated effort to copy things around?